File: jimeng-ai-webapp/backend/app/schemas/article.py
"""文章相关请求与响应模型"""
from typing import Optional, List
from pydantic import BaseModel, Field


class ArticleCreate(BaseModel):
    """创建文章请求"""
    title: str = Field(..., min_length=1, max_length=200, description="文章标题")
    content: Optional[str] = Field(None, description="Markdown格式正文")
    summary: Optional[str] = Field(None, max_length=500, description="摘要")
    category_id: Optional[str] = Field(None, description="分类ID")
    tag_ids: Optional[List[str]] = Field(None, description="标签ID列表")
    cover_image_path: Optional[str] = Field(None, description="封面图路径")
    status: str = Field(default="draft", description="文章状态: draft/published")


class ArticleUpdate(BaseModel):
    """更新文章请求"""
    title: Optional[str] = Field(None, min_length=1, max_length=200, description="文章标题")
    content: Optional[str] = Field(None, description="Markdown格式正文")
    summary: Optional[str] = Field(None, max_length=500, description="摘要")
    category_id: Optional[str] = Field(None, description="分类ID")
    tag_ids: Optional[List[str]] = Field(None, description="标签ID列表")
    cover_image_path: Optional[str] = Field(None, description="封面图路径")
    status: Optional[str] = Field(None, description="文章状态")


class ArticleAutoSave(BaseModel):
    """自动保存请求"""
    title: str = Field(..., min_length=1, max_length=200)
    content: Optional[str] = None
    summary: Optional[str] = None


class ArticleGenerate(BaseModel):
    """AI生成文章请求"""
    query: str = Field(..., min_length=1, max_length=500, description="文章主题/关键词")
    source_ids: Optional[List[str]] = Field(None, description="选中的素材ID列表")
    style: Optional[str] = Field(None, description="写作风格")
    generate_images: bool = Field(default=False, description="是否同时生成配图")


class ArticleResponse(BaseModel):
    """文章响应"""
    id: str
    title: str
    content: Optional[str] = None
    summary: Optional[str] = None
    status: str
    category_id: Optional[str] = None
    category_name: Optional[str] = None
    cover_image_path: Optional[str] = None
    word_count: int = 0
    view_count: int = 0
    tags: List[dict] = []
    created_at: Optional[str] = None
    updated_at: Optional[str] = None


class ArticleListItem(BaseModel):
    """文章列表项"""
    id: str
    title: str
    summary: Optional[str] = None
    status: str
    category_name: Optional[str] = None
    cover_image_path: Optional[str] = None
    word_count: int = 0
    tags: List[dict] = []
    created_at: Optional[str] = None


class ArticleVersionResponse(BaseModel):
    """文章版本响应"""
    id: str
    article_id: str
    version_number: int
    title: str
    content: Optional[str] = None
    change_summary: Optional[str] = None
    created_at: Optional[str] = None


class CategoryCreate(BaseModel):
    """创建分类请求"""
    name: str = Field(..., min_length=1, max_length=50, description="分类名称")
    description: Optional[str] = Field(None, max_length=200, description="分类描述")
    sort_order: int = Field(default=0, description="排序")


class CategoryUpdate(BaseModel):
    """更新分类请求"""
    name: Optional[str] = Field(None, min_length=1, max_length=50)
    description: Optional[str] = Field(None, max_length=200)
    sort_order: Optional[int] = None


class CategoryResponse(BaseModel):
    """分类响应"""
    id: str
    name: str
    description: Optional[str] = None
    sort_order: int = 0
    article_count: int = 0
    created_at: Optional[str] = None


class TagCreate(BaseModel):
    """创建标签请求"""
    name: str = Field(..., min_length=1, max_length=30, description="标签名")
    color: str = Field(default="#6366F1", max_length=7, description="标签颜色")


class TagUpdate(BaseModel):
    """更新标签请求"""
    name: Optional[str] = Field(None, min_length=1, max_length=30)
    color: Optional[str] = Field(None, max_length=7)


class TagResponse(BaseModel):
    """标签响应"""
    id: str
    name: str
    color: str = "#6366F1"
    article_count: int = 0
    created_at: Optional[str] = None

File: jimeng-ai-webapp/frontend/src/api/article.ts
import request from './request'

// ========== 类型定义 ==========

export interface Article {
  id: string
  title: string
  content?: string
  summary?: string
  status: string
  category_id?: string
  category_name?: string
  cover_image_path?: string
  word_count: number
  view_count: number
  tags: Tag[]
  created_at?: string
  updated_at?: string
}

export interface ArticleListItem {
  id: string
  title: string
  summary?: string
  status: string
  category_name?: string
  cover_image_path?: string
  word_count: number
  tags: Tag[]
  created_at?: string
}

export interface Tag {
  id: string
  name: string
  color: string
  article_count?: number
}

export interface Category {
  id: string
  name: string
  description?: string
  sort_order: number
  article_count?: number
}

export interface ArticleVersion {
  id: string
  article_id: string
  version_number: number
  title: string
  content?: string
  change_summary?: string
  created_at?: string
}

// ========== 文章API ==========

export async function getArticleList(params: {
  page?: number
  page_size?: number
  status?: string
  category_id?: string
  keyword?: string
}) {
  return request.get('/articles', { params })
}

export async function getArticle(id: string) {
  return request.get(`/articles/${id}`)
}

export async function createArticle(data: {
  title: string
  content?: string
  summary?: string
  category_id?: string
  tag_ids?: string[]
  cover_image_path?: string
  status?: string
}) {
  return request.post('/articles', data)
}

export async function updateArticle(id: string, data: {
  title?: string
  content?: string
  summary?: string
  category_id?: string
  tag_ids?: string[]
  cover_image_path?: string
  status?: string
}) {
  return request.put(`/articles/${id}`, data)
}

export async function deleteArticle(id: string) {
  return request.delete(`/articles/${id}`)
}

export async function autoSaveArticle(id: string, data: {
  title: string
  content?: string
  summary?: string
}) {
  return request.post(`/articles/${id}/auto-save`, data)
}

export async function getArticleVersions(id: string) {
  return request.get(`/articles/${id}/versions`)
}

export async function restoreArticleVersion(articleId: string, versionId: string) {
  return request.post(`/articles/${articleId}/versions/${versionId}/restore`)
}

export async function saveArticleSources(articleId: string, sources: any[]) {
  return request.post(`/articles/${articleId}/sources`, sources)
}

export async function getArticleSources(articleId: string) {
  return request.get(`/articles/${articleId}/sources`)
}

export async function generateArticle(articleId: string, data: {
  query: string
  source_ids?: string[]
  style?: string
}) {
  return request.post(`/articles/${articleId}/generate`, data)
}

// ========== 分类API ==========

export async function getCategories() {
  return request.get('/categories')
}

export async function createCategory(data: { name: string; description?: string; sort_order?: number }) {
  return request.post('/categories', data)
}

export async function updateCategory(id: string, data: { name?: string; description?: string; sort_order?: number }) {
  return request.put(`/categories/${id}`, data)
}

export async function deleteCategory(id: string) {
  return request.delete(`/categories/${id}`)
}

// ========== 标签API ==========

export async function getTags() {
  return request.get('/tags')
}

export async function createTag(data: { name: string; color?: string }) {
  return request.post('/tags', data)
}

export async function updateTag(id: string, data: { name?: string; color?: string }) {
  return request.put(`/tags/${id}`, data)
}

export async function deleteTag(id: string) {
  return request.delete(`/tags/${id}`)
}
